include/revng/Yield/Assembly/Assembly.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <optional>
+#include <type_traits>
+
+#include "llvm/ADT/SmallVector.h"
+
+#include "revng/ADT/KeyedObjectTraits.h"
+#include "revng/ADT/SortedVector.h"
+#include "revng/Support/MetaAddress.h"
+
+namespace assembly {
+
+struct Instruction {
```

Comment:
Did you consider using tuple tree generator? It'd enable us to ship to clients a YAML document with this information (as opposed to HTML) with no effort.
I understand it might seem overkill, but I'd also appreciate your reasoning why you think it's excessive, because we might want to reduce friction to the point this is no longer a problem.

Reply:
There are some non-trivial changes happening when this stuff gets converted into html (like function/basic_block resolution and the line), so I don't think this is suitable for the client.
 
On the other hand, I want to have access to this stuff when building graphs though. So switching to the tuple tree generator might be a good idea either way.

Reply:
If you think the client would benefit from getting yaml over html, I'm all for it - but that yaml shouldn't look much like this data-structure.

Reply:
In practice it wouldn't broaden it that much - the target yaml contains pretty much all the same things html contains now but ordered slightly differently. But yeah, some other day.